src/components/lib/kernel/cos_kernel_api.c

```diff
@@ -688,7 +694,7 @@ __page_bump_alloc(struct cos_compinfo *ci, size_t sz)
 	for (heap_cursor = heap_vaddr; heap_cursor < heap_limit; heap_cursor += PAGE_SIZE) {
 		vaddr_t umem;
 
-		umem = __umem_bump_alloc(ci);
+		umem = shared ? __kmem_bump_alloc(ci) : __umem_bump_alloc(ci);
```

Comment:
This looks like it is building toward DCBs being Virt mem first. I hope not. I hope they are a kernel data-structure first, then mapped into virt mem.